### PR TITLE
Add SmallestAI Pulse STT and Lightning TTS to pipecat cascade pipeline

### PR DESCRIPTION
**Pull request by weiz9**
## Summary
- Wires pipecat's `SmallestSTTService` (Pulse) and `SmallestTTSService` (Lightning v3.1 / v3.1 Pro) into `create_stt_service`/`create_tts_service`, so the CASCADE pipeline (STT -> LLM -> TTS) can use Smallest as a provider alongside the existing SmallestAI Hydra S2S integration.
- Requires `pipecat-ai>=1.4.0` (already the pinned minimum in `pyproject.toml`).
- Bumped `simulation_version` (2.0.1 -> 2.0.2) since this touches `src/eva/assistant/`.
- Added `smallest` to the STT/TTS provider enums in `.env.example`.

## Test plan
- [x] `pytest tests/unit/assistant/test_services.py -v` — new smallest STT/TTS unit tests pass, including the default-voice-preservation edge case (voice_id omitted -> pipecat's per-model default voice applies).
- [x] `pytest tests/unit -q` — full unit suite passes (1846 passed).
- [x] `ruff check` / `ruff format --check` clean; `mypy` shows no new errors beyond the pre-existing `api_key: Any | None` pattern shared by every other provider in this file.
- [x] End-to-end `EVA_DEBUG=true eva --domain airline` with `stt=smallest` (pulse), `tts=smallest` (lightning_v3.1_pro), `llm=gpt-5.4`: 3/3 conversations succeeded. Verified valid non-silent 24kHz audio, clean multi-turn transcripts, zero error frames from `SmallestSTTService`/`SmallestTTSService` in pipecat logs, and reasonable STT WER (~9%).

### CODE REVIEW

File: src/eva/__init__.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 
 # Bump simulation_version when changes affect benchmark outputs (agent code,
 # user simulator, orchestrator, simulation prompts, agent configs, tool mocks).
-simulation_version = "2.0.1"
+simulation_version = "2.0.2"
```
Comment (fanny-riols) [inline src/eva/__init__.py]: I don't think we need to bump the version here. You are only adding support to smallestai, so it doesn't affect the simulator.
```suggestion
simulation_version = "2.0.1"
```

File: src/eva/assistant/pipeline/services.py [hunk 2]
```diff
@@ -279,6 +281,21 @@ def create_stt_service(
             )
         return stt_service
 
+    elif model_lower == "smallest":
+        logger.info(f"Using Smallest Pulse STT: {params['model']}")
```
Comment (fanny-riols) [inline src/eva/assistant/pipeline/services.py]: The model name might change with time.
```suggestion
        logger.info(f"Using Smallest STT: {params['model']}")
```

File: src/eva/assistant/pipeline/services.py [hunk 3]
```diff
@@ -500,6 +517,25 @@ def create_tts_service(
 
         return openai_tts
 
+    elif model_lower == "smallest":
+        logger.info(f"Using Smallest Lightning TTS: {params['model']}")
```
Comment (fanny-riols) [inline src/eva/assistant/pipeline/services.py]:
```suggestion
        logger.info(f"Using Smallest TTS: {params['model']}")
```

**Review by fanny-riols — APPROVED** [review]
